client/alloc_runner.go

```diff
@@ -584,6 +584,8 @@ func (r *AllocRunner) handleDestroy() {
 			r.logger.Printf("[ERR] client: failed to destroy state for alloc '%s': %v",
 				r.alloc.ID, err)
 		}
+	case <-r.updateCh:
```

Comment:
Should we put this select in a for loop with a return in the `<-r.destroyCh` case?